vm/rust/src/lib.rs

```diff
@@ -205,12 +205,56 @@ pub extern "C" fn cairoVMExecute(
                 JunoAppendGasConsumed(
                     reader_handle,
                     felt_to_byte_array(&t.actual_fee.0.into()).as_ptr(),
-                )
+                );
+
+                append_execution_info(
+                    reader_handle,
+                    t.into(),
+                );
             },
         }
     }
 }
 
+fn transaction_from_api(
+    tx: StarknetApiTransaction,
+    contract_class: Option<ContractClass>,
+) -> Result<Transaction, String> {
+    match tx {
+        StarknetApiTransaction::L1Handler(l1_handler) => {
+            Ok(Transaction::L1HandlerTransaction(L1HandlerTransaction {
+                tx: l1_handler,
+                // for now just passing non zero value
+                paid_fee_on_l1: Fee(1 as u128),
+            }))
+        }
+        StarknetApiTransaction::Declare(declare) => {
+            let declare_tx = DeclareTransaction::new(
+                declare,
+                contract_class.ok_or("Declare should be created with a ContractClass")?,
+            ).map_err(|e| e.to_string())?;
+            Ok(Transaction::AccountTransaction(AccountTransaction::Declare(declare_tx)))
+        }
+        StarknetApiTransaction::DeployAccount(deploy_account) => {
+            Ok(Transaction::AccountTransaction(AccountTransaction::DeployAccount(deploy_account)))
+        }
+        StarknetApiTransaction::Invoke(invoke) => {
+            Ok(Transaction::AccountTransaction(AccountTransaction::Invoke(invoke)))
+        }
+        StarknetApiTransaction::Deploy(deploy) => {
```

Comment:
Let's not copy and paste `Transaction::from_api` here. 

Instead let's check for the cases that `Transaction::from_api` panic and make sure we error out early don't call it in those cases.

Reply:
Ok I added call to `Transaction::from_api`, also decided to make things clean to keep validation in existing function `transaction_from_api(...)`